packages/x-charts/src/BarChart/seriesConfig/bar/getNextFocusedItem.ts

```diff
@@ -0,0 +1,42 @@
+import {
+  getNextIndexFocusedItem,
+  getPreviousIndexFocusedItem,
+  getPreviousSeriesFocusedItem,
+  getNextSeriesFocusedItem,
+} from '../../../internals/commonNextFocusItem';
+import type { GetNextFocusedItem } from '../../../internals/plugins/models/seriesConfig/getNextFocusedItem.types';
+
+const outSeriesTypes: Set<'bar' | 'line' | 'scatter'> = new Set(['bar', 'line', 'scatter']);
+
+const getNextFocusedItem: GetNextFocusedItem<'bar'> = (currentItem, event, state) => {
+  switch (event.key) {
+    case 'ArrowRight':
+      return getNextIndexFocusedItem<'bar', 'bar' | 'line' | 'scatter'>(
+        currentItem,
+        outSeriesTypes,
+        state,
+      );
+    case 'ArrowLeft':
+      return getPreviousIndexFocusedItem<'bar', 'bar' | 'line' | 'scatter'>(
+        currentItem,
+        outSeriesTypes,
+        state,
+      );
+    case 'ArrowDown':
+      return getPreviousSeriesFocusedItem<'bar', 'bar' | 'line' | 'scatter'>(
+        currentItem,
+        outSeriesTypes,
+        state,
+      );
+    case 'ArrowUp':
+      return getNextSeriesFocusedItem<'bar', 'bar' | 'line' | 'scatter'>(
+        currentItem,
+        outSeriesTypes,
+        state,
+      );
+    default:
+      return null;
+  }
+};
```

Comment:
if the api is the same for every item we could move the abstraction up so we don't need to split it here. We just care about the returned function

```tsx
const getNextFocusedItem: GetNextFocusedItem<'bar'> = (event) => {
  switch (event.key) {
    case 'ArrowRight':
      return getNextIndexFocusedItem
    case 'ArrowLeft':
      return getPreviousIndexFocusedItem
    case 'ArrowDown':
      return getPreviousSeriesFocusedItem
    case 'ArrowUp':
      return getNextSeriesFocusedItem
    default:
      return null;
  }
};
```

then `outSeriesTypes` is just a const, can be part of the series config as well.